core/src/main/java/org/apache/iceberg/util/SerializationUtil.java

```diff
@@ -43,6 +45,16 @@ private SerializationUtil() {
     }
   }
 
+  public static byte[] serializeToBytesWithHadoopConfig(Object obj) {
```

Comment:
Why introduce a separate method here rather than supporting `HadoopConfigurable` in `serializeToBytes`? It seems less useful if use of `HadoopConfigurable` isn't automatic and you need to remember to call the right method.

Reply:
The original thinking was that the use of `org.apache.iceberg.hadoop.SerializableConfiguration` was not always a default. `SerializableTable` uses a map based serializer, Spark also has its own serializer. So I don't want the user to blindly assume that `serializeToBytes` would take care of the Hadoop configuration in all places. If we are making it a default, then I will add a documentation for `serializeToBytes` to make this clear.

Reply:
I see what you mean. In that case, maybe it would make more sense to allow passing the `serializeConfWith` function into `serializeToBytes` as an option? If you don't pass it, then we could use the current `SerializableConfiguration` as the default.